modules/jupyter-widget/src/binary-transport.js

```diff
@@ -143,22 +127,19 @@ function makeBinaryAccessorProps(accessors, layerId, dataBuffer) {
   return dataAccessorProps;
 }
 
-function processDataBuffer({currentLayers, dataBuffer}) {
-  const updatedLayers = [];
-  const updateLayerIds = Object.keys(dataBuffer);
-  for (const layerId of updateLayerIds) {
-    const currentLayer = getLayer(currentLayers, layerId);
-    const newLayerData = constructData(dataBuffer, layerId);
-    const {LayerConstructor, layerProps} = getCurrentLayerProps(currentLayer);
-    const accessors = getAccessorNamesFrom(dataBuffer, layerId);
-    const accessorProps = makeBinaryAccessorProps(accessors, layerId, dataBuffer);
-    const combinedProps = {...layerProps, ...accessorProps};
-    combinedProps.data = newLayerData;
-    const layer = new LayerConstructor({...combinedProps});
-    updatedLayers.push(layer);
+function processDataBuffer({dataBuffer, jsonProps}) {
+  const convertedJsonProps = jsonConverter(jsonProps);
+  for (let i = 0; i < convertedJsonProps.layers.length; i++) {
+    const layer = convertedJsonProps.layers[i];
+    layer.data = constructData(dataBuffer, layer.id);
+    // use https://deck.gl/#/documentation/deckgl-api-reference/layers/layer?section=state-object- instead
+    const accessors = getAccessorNamesFrom(dataBuffer, layer.id);
```

Comment:
You do not need to specify accessors when using binary data. They will not be called.

Reply:
They turn the accessor values into functions like [in the docs here](https://deck.gl/#/documentation/developer-guide/performance-optimization?section=use-binary-data).

```
    getFillColor: (object, {index, data, target}) => {
      return data.src.subarray(index * 6 + 3, index * 6 + 6);
    }
```

Reply:
There are multiple examples in that doc. You should follow the “directly supply attributes” one.

Reply:
There are multiple examples in that doc. You should follow the “directly supply attributes” one.